### PR TITLE
Disable some FileWatching tests on x86_64 apple

### PR DESCRIPTION
**Pull request by gbaraldi**
Macos does some weird event reordoring which means these tests fail often polluting CI output

### CODE REVIEW

**Comment by DilumAluthge** [comment]
Red https://github.com/JuliaLang/julia/issues/46185, right?

**Comment by gbaraldi** [comment]
Hopefully yeah

**Review by Keno — APPROVED** [review]

File: stdlib/FileWatching/test/runtests.jl [hunk 1]
```diff
@@ -369,60 +369,62 @@ test_monitor_wait_poll()
...
-            @test pop!(changes)[2] == FileWatching.FileEvent(FileWatching.UV_RENAME)
-        end
-        p = pop!(changes)
-        if !Sys.isapple()
-            @test p == (F_PATH * "~" => FileWatching.FileEvent(FileWatching.UV_RENAME))
-        end
-        while changes[end][1] == F_PATH * "~"
-            @test pop!(changes)[2] == FileWatching.FileEvent(FileWatching.UV_RENAME)
+
+if !((Sys.ARCH == :x86_64) && (Sys.isapple())) #These tests tend to fail a lot on x86-apple
```
Comment (Keno) [inline stdlib/FileWatching/test/runtests.jl:373]: Whoops, I may have been a little overeager. These fail on aarch64 also, so should be disabled everywhere.

Comment (gbaraldi) [reply stdlib/FileWatching/test/runtests.jl:373]: I actually haven't seen this fail in FileWatching in aarch64, the only one I've seen was a failure in the PIDFile test.

Comment (Keno) [reply stdlib/FileWatching/test/runtests.jl:373]: Ok, but the moment we see it, the test is gone ;)

**Pull request merged**